addons/utm/models/utm_campaign.py

```diff
@@ -25,6 +27,26 @@ class UtmCampaign(models.Model):
     is_auto_campaign = fields.Boolean(default=False, string="Automatically Generated Campaign", help="Allows us to filter relevant Campaigns")
     color = fields.Integer(string='Color Index')
 
+    _sql_constraints = [
+        ('unique_name', 'UNIQUE(name)', 'The name must be unique'),
+    ]
+
+    @api.depends('title')
+    def _compute_name(self):
+        new_names = self.env['utm.mixin']._get_unique_names(self._name, [c.title for c in self])
```

Comment:
Shouldn't we skip if we have a name and title is void ?

Reply:
no, because we expect the returned values to have the same size as self (`_get_unique_names` care about falsy values)